distributed/tests/test_scheduler.py

```diff
@@ -3978,3 +3978,16 @@ def test_runspec_regression_sync(loop):
         # serialization errors that result in KilledWorker
         with pytest.raises(IndexError):
             overlapped.compute()
+
+
+@gen_cluster(config={"distributed.scheduler.allowed-failures": 666})
+async def test_KilledWorker_informative_message(s, a, b):
+    ws = s.workers[a.address].clean()
+    ex = KilledWorker("foo-bar", ws, s.allowed_failures)
+    with pytest.raises(KilledWorker) as excinfo:
+        raise ex
+    msg = str(excinfo.value)
```

Comment:
```suggestion
    msg = str(ex)
```
Is the raising necessary to test?

Reply:
Probably not. It wasn't immediately clear to me if `__repr__` or `__str__` was going to be what users see (I always forget which one is supposed to be unambiguous and which is more meaningful for users). Explicitly raising made the intent of the test more clear to me -- though I certainly could remove the explicit `raise` if needed

Reply:
Just checking -- are you okay with the current changes or would you like me to remove the explicit `raise`?

Reply:
Keeping the explicit `raise` is fine with me